opal/mca/memory/patcher/configure.m4

```diff
@@ -38,53 +38,18 @@ AC_DEFUN([MCA_opal_memory_patcher_CONFIG],[
 
     OPAL_VAR_SCOPE_PUSH([memory_patcher_have___curbrk memory_patcher_have___mmap memory_patcher_have___syscall memory_patcher_have___mmap_prototype memory_patcher_have___syscall_prototype])
 
-    memory_patcher_have___curbrk=0
-    memory_patcher_have___mmap=0
-    memory_patcher_have___mmap_prototype=0
-    memory_patcher_have___syscall=0
-    memory_patcher_have___syscall_prototype=0
+    AC_CHECK_FUNCS([__curbrk])
 
-    AC_MSG_CHECKING([for __curbrk symbol])
-    AC_LINK_IFELSE([AC_LANG_PROGRAM([extern char *__curbrk;],[char *tmp = __curbrk;])],
-                   [AC_MSG_RESULT([yes])
-                    memory_patcher_have___curbrk=1],
-                   [AC_MSG_RESULT([no])])
-    AC_DEFINE_UNQUOTED([OPAL_MEMORY_PATCHER_HAVE___CURBRK], [$memory_patcher_have___curbrk],
-                       [Whether the glibc __curbrk exists])
+    AC_CHECK_HEADERS([linux/mman.h sys/syscall.h])
```

Comment:
note that was already there, i just moved it.

from ```opal/mca/memory/patcher/memory_patcher_component.c```
```c
#include <sys/mman.h>
#if defined(HAVE_SYS_SYSCALL_H)
#include <sys/syscall.h>
#endif
#if defined(HAVE_LINUX_MMAN_H)
#include <linux/mman.h>
#endif
```

bottom line, the current code requires it, and i do not know the details.